### PR TITLE
HPCC-31966 Add internal spans for graph execution within roxie

### PR DESCRIPTION
**Pull request by ghalliday**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: roxie/ccd/ccdcontext.cpp [hunk 1]
```diff
@@ -1658,6 +1658,7 @@ class CRoxieContextBase : implements IRoxieAgentContext, implements ICodeContext
         }
         else
         {
+            OwnedSpanScope graphScope = queryActiveSpan()->createInternalSpan(name);
```
Comment (rpastrana) [inline roxie/ccd/ccdcontext.cpp:1661]: This seems fine, looking at a sample roxie trace visualization, where would this span reside, is the name good enough, and are there any attributes we should include in the graph span, or are we simply providing the graph's name as reference, and the timing for completeness? 

image

**Review by rpastrana — APPROVED** [review]
@ghalliday looks fine, left some questions to consider

File: roxie/ccd/ccdcontext.cpp (hunk 1 above)
Comment (ghalliday) [reply roxie/ccd/ccdcontext.cpp:1661]: It will occur as a child of the main service.  The soapcall activites would be children.

Comment (ghalliday) [reply roxie/ccd/ccdcontext.cpp:1661]: We can revisit at a later date to decide if it is worth adding any extra information, but the name and start/duration is what I would be interested in.

Comment (rpastrana) [reply roxie/ccd/ccdcontext.cpp:1661]: Let's get it merged and out into the wild, I'll ask for feedback from our users and we can improve later.

**Comment by ghalliday** [comment]
@rpastrana I reimplemented it to use the queryThreadedActiveSpan() function instead.  The reason being that I think we can remove queryActiveSpan() from the context now that is implemented, and this can be an initial test.

**Review by rpastrana — APPROVED** [review]
@ghalliday seems fine

**Review by rpastrana — COMMENTED** [review]
After approving I wondered if this new span is now set as the active span at the thread level

**Comment by ghalliday** [comment]
> After approving I wondered if this new span is now set as the active span at the thread level
Do you think that is a problem, or correct?

It should be - which means that at threads that are started will now have the correct parent span.

**Comment by ghalliday** [comment]
@rpastrana see question

**Comment by rpastrana** [comment]
> @rpastrana see question

Looking around for a question from you, but I only see the response to my question. 
Your answer is what I expected. The new span is set as active by virtue of OwnedSpanScope::OwnedSpanScope() and I agree, it's the correct approach.

**Pull request merged**